pkg/chunk/cached_store.go

```diff
@@ -665,33 +668,38 @@ func (store *cachedStore) load(key string, page *Page, cache bool, forceCache bo
 	var in io.ReadCloser
 	tried := 0
 	start := time.Now()
-	// it will be retried outside
-	for err != nil && tried < 2 {
-		time.Sleep(time.Second * time.Duration(tried*tried))
-		if tried > 0 {
-			logger.Warnf("GET %s: %s; retrying", key, err)
-			store.objectReqErrors.Add(1)
-			start = time.Now()
-		}
-		in, err = store.storage.Get(key, 0, -1)
-		tried++
-	}
 	var n int
 	var buf []byte
-	if err == nil {
-		if compressed {
-			c := NewOffPage(needed)
-			defer c.Release()
-			buf = c.Data
-		} else {
-			buf = page.Data
+	page.Acquire()
+	err = utils.WithTimeout(func() error {
+		defer page.Release()
+		// it will be retried outside
+		for err != nil && tried < 2 {
+			time.Sleep(time.Second * time.Duration(tried*tried))
+			if tried > 0 {
+				logger.Warnf("GET %s: %s; retrying", key, err)
+				store.objectReqErrors.Add(1)
+				start = time.Now()
+			}
+			in, err = store.storage.Get(key, 0, -1)
```

Comment:
only wrap the `Get` method with `GetTimeout`?

Reply:
Get() return reader, it may hang inside Read()